src/text_shaper/open_shaper.cpp

```diff
@@ -677,8 +678,14 @@ void open_shaper::shape(font_key font,
         logMessage.append("Using font: key={}, path=\"{}\"\n", font, identifierOf(fontInfo.primary));
     }
 
-    if (_d->tryShapeWithFallback(
-            font, fontInfo, hbBuf, hbFont, script, presentation, codepoints, clusters, result))
+    auto shapingFunction = [&](auto&&... args) {
+        if (useFontFallback)
```

Comment:
Hmmm 🤔 
Is this really useful, to be able to completely turn off font fallback?

Even if, we should not do something like `if (useFontFallback) ... else ...` in the rendering hot path, but rely on function pointers / virtual dispatch instead. I know this sounds like a micro optimization, but would be cleaner IMHO on top of it, too.

Ignoring that, I think we could instead promote the `mock` font locator instead, because there you can explicitly declare how many fonts (font files) to pick in order for fallback. If one would not specify any more than a single font, that would be the same as having no font fallback at all.

But none of it is actually addressing the point in #225.

We could instead work on embedding fonts (Nerd Fonts and standard emoji fonts), which would help people like @eproxus to not require font patching (nerd font), or always come with the same experience when enabling something like `use_builtin_emoji: true` (or `emoji: builtin` or not specifying `emoji:` key at all, indicating that the builtin should be used)

Reply:
I updated PR, now we configure fallback on the stage of font_locator and also can specify list of fonts that use for fallback